### PR TITLE
tailcfg: add Hostinfo.HowUnequal method

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -577,6 +577,68 @@ func (h *Hostinfo) Equal(h2 *Hostinfo) bool {
...
 }
 
+// HowUnequal returns a list of paths through Hostinfo where h and h2 differ.
+// If they differ in nil-ness, the path is "nil", otherwise the path is like
+// "ShieldsUp" or "NetInfo.nil" or "NetInfo.PCP".
+func (h *Hostinfo) HowUnequal(h2 *Hostinfo) (path []string) {
+	return appendStructPtrDiff(nil, "", reflect.ValueOf(h), reflect.ValueOf(h2))
+}
+
+func appendStructPtrDiff(base []string, pfx string, p1, p2 reflect.Value) (ret []string) {
```
Comment (crawshaw) [inline tailcfg/tailcfg.go:587]: 🤒

**Review by crawshaw — APPROVED** [review]
Want to put some stuff in NetInfo in your tests to test pointer traversal? (Also compare a hostinfo with a netinfo, and one with a nil netinfo)

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (bradfitz) [reply tailcfg/tailcfg.go:587]: 😊

**Comment by bradfitz** [comment]
> Want to put some stuff in NetInfo in your tests to test pointer traversal? (Also compare a hostinfo with a netinfo, and one with a nil netinfo)

I think I have all those tests already?

**Comment by crawshaw** [comment]
I suppose so! Lots more potential edge cases with dereferencing via reflect, but we'll find them fast if they're there.
